react/features/virtual-background/components/VirtualBackgroundDialog.js

```diff
@@ -54,23 +59,97 @@ type Props = {
  * @returns {ReactElement}
  */
 function VirtualBackground({ dispatch, t }: Props) {
+    const localImages = jitsiLocalStorage.getItem('virtualBackgrounds');
+    const [ storedImages, setStoredImages ] = useState((localImages && JSON.parse(localImages)) || []);
+    const [ loading, isloading ] = useState(false);
+
+    const toDataURL = url =>
+        fetch(url)
```

Comment:
can you please write this as an async function and use await instead? Also, take it out of the component, it's generic and can go in functions.js